Sources/Bond/UIKit/UICollectionView.swift

```diff
@@ -136,6 +134,7 @@ public extension SignalProtocol where Element: DataSourceEventProtocol, Element.
                     applyEventOfKind(event.kind)
                 }
             }
+            _ = collectionView.numberOfSections // Hack to immediately apply changes
```

Comment:
Hey @killev,

This PR looks great. Could you elaborate why exactly is this needed? Is it only for tests to pass or there is something that actually does not work without this?

Reply:
Briefly - it's rather UICollectionView's issue. It crashes if you change it before it appears on screen (viewDidAppear). UITableView works great with the same

Many libraries suffer from the same issue,  IGListKit for instance. 

But far as Reactive approach is driven by data changes, and bindings should work despite the fact a view is shown or hidden. So we can't ignore this.

Try a simple example here:
https://gist.github.com/killev/b802c213a1e1a9d05853ca7b869bcc28

Reply:
I've been investigating this in depth. Looks like the issue happens when collection view is not in window hierarchy and we do partial updates. 

I think the right solution is to put the following on top of our method:

```swift
guard collectionView.window != nil else {
    collectionView.reloadData()
    return
}
``` 

If we are not in a window, just reload, don't do any partial updates. 

I'm going to merge your changes and add this. Thanks a lot for figuring the issue out!